components/hierarchical-view/hierarchical-view-mixin.js

```diff
@@ -472,4 +505,28 @@ export const HierarchicalViewMixin = superclass => class extends superclass {
 		this.__resizeObserver.disconnect();
 		this.__resizeObserver.observe(content);
 	}
+
+	__updateFocusables(view, sourceAttribute, targetAttribute) {
+		const rootView = this.getRootView();
+		let node = getComposedParent(view);
+		while (node && node !== rootView) {
+			if (node.nodeType === Node.ELEMENT_NODE) {
+				const tabIndex = node.getAttribute(sourceAttribute);
+				if (tabIndex !== null) {
+					node.setAttribute(targetAttribute, tabIndex);
+					node.removeAttribute(sourceAttribute);
+				}
+			}
+			node = getComposedParent(node);
+		}
+	}
+
+	__updateFocusablesHide(view) {
+		this.__updateFocusables(view, 'd2l-hierarchical-view-tabindex', 'tabindex');
```

Comment:
If these elements could include native elements, would it be better to throw this into a `data-*` attribute? Just to avoid having linters and HTML validators complaining at us for using unsupported attribute names on `<div>`s and such.

Reply:
Thanks, yeah,  This was meant to use a data attribute.